src/DependencyManagement/DependencyManagementUtils.cs

```diff
@@ -0,0 +1,164 @@
+                    RemoveFileInfo(fileInfo);
+                }
+
+                foreach (var subDirectory in directoryInfo.GetDirectories())
+                {
+                    subDirectory.Delete(true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Deletes the given fileInfo.
+        /// </summary>
+        internal static void RemoveFileInfo(FileInfo fileInfo)
+        {
+            try
+            {
+                // Remove any problematic file attributes.
+                fileInfo.Attributes = fileInfo.Attributes &
+                                      ~(FileAttributes.Hidden | FileAttributes.ReadOnly | FileAttributes.System);
+                fileInfo.Delete();
+            }
+            catch
+            {
+                // It is ok to ignore errors from trying to delete the file.
+            }
+        }
+
+        /// <summary>
+        /// Returns the latest module version from the PSGallery for the given module name and major version.
+        /// </summary>
+        internal static string GetModuleLatestSupportedVersion(string moduleName, string majorVersion)
+        {
+            Uri address = new Uri("https://www.powershellgallery.com/api/v2/FindPackagesById()?id='" + moduleName + "'");
+            int configuredRetries = 3;
+            int noOfRetries = 1;
+
+            string requestContent = null;
+            while (noOfRetries <= configuredRetries)
+            {
+                try
+                {
+                    HttpWebRequest request = WebRequest.Create(address) as HttpWebRequest;
+                    using (HttpWebResponse response = request?.GetResponse() as HttpWebResponse)
+                    {
+                        var stream = response?.GetResponseStream();
+                        if (stream != null)
+                        {
+                            StreamReader reader = new StreamReader(stream);
+                            requestContent = reader.ReadToEnd();
+                        }
+                    }
+
+                    break;
+                }
+                catch (Exception ex)
+                {
+                    WebException webEx = ex as WebException;
+                    if (webEx == null || noOfRetries >= configuredRetries)
+                    {
+                        throw;
+                    }
+
+                    // Only retry the web exception
+                    if (IsRetriable(webEx))
+                    {
+                        noOfRetries++;
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+            }
+
+            if (string.IsNullOrEmpty(requestContent))
+            {
+                return null;
+            }
+
+            // Load up the XML response
+            XmlDocument doc = new XmlDocument();
+            using (XmlReader reader = XmlReader.Create(new StringReader(requestContent)))

```

Comment:
I originally had it like that, but I was getting a warning from the compiler for a possible null assigned, please see below. So, I will leave it as is for now.

![image](https://user-images.githubusercontent.com/12720858/54396884-c319af80-4671-11e9-879b-98941edf05a2.png)

Reply:
What is that warning message?
You have that warning probably because `response?.GetResponseStream()` can result in `null` in case `response == null`.

Reply:
The stream `requestContent` is not really necessary, you can use the `stream` from response directly.